test/network_analysis/network_properties.jl

```diff
@@ -649,286 +320,103 @@ let
+    @test count(k -> !isequal(k, 0), K) == length(reactions(MAPK))
+    K = fluxmat(MAPK; sparse = true)
+    @test Catalyst.issparse(K)
+    
+    A_k = Catalyst.laplacianmat(MAPK)
+    @test all(col -> sum(col) == 0, eachcol(A_k))
+    A_k = Catalyst.laplacianmat(MAPK; sparse = true)
+    @test Catalyst.issparse(A_k)
 
-# Fails because linkage deficiencies do not sum to total deficiency
-let 
-    rn = @reaction_network begin
-        (k1, k2), A <--> 2A
-        (k3, k4), A + B <--> C
-        (k5, k6), C <--> B 
-    end
+    S = netstoichmat(MAPK); Y = complexstoichmat(MAPK)
+    @test isequal(S*K, Y*A_k)
 
-    @test Catalyst.satisfiesdeficiencyone(rn) == false
-end
+    eqs = Catalyst.assemble_oderhs(MAPK, specs)
+    @test all(iszero, simplify(eqs - S*K*Φ))
+    @test all(iszero, simplify(eqs - Y*A_k*Φ))
 
-# Fails because a linkage class has deficiency two
-let 
-    rn = @reaction_network begin
-        k1, 3A --> A + 2B
-        k2, A + 2B --> 3B
-        k3, 3B --> 2A + B
-        k4, 2A + B --> 3A
-    end
+    # Test using numbers
+    k = rand(rng, numparams(MAPK))
+    ratevec = collect(zip(parameters(MAPK), k))
+    ratemap = Dict(ratevec)
+    ratetup = Tuple(ratevec)
 
-    @test Catalyst.satisfiesdeficiencyone(rn) == false
-end
+    @test Catalyst.fluxmat(MAPK, ratemap) == Catalyst.fluxmat(MAPK, ratevec) == Catalyst.fluxmat(MAPK, ratetup)
+    K = Catalyst.fluxmat(MAPK, ratemap; sparse = true)
+    @test Catalyst.issparse(K)
+    
+    A_k = Catalyst.laplacianmat(MAPK, ratemap)
+    @test all(col -> sum(col) == 0, eachcol(A_k))
+    A_k = Catalyst.laplacianmat(MAPK, ratemap; sparse = true)
+    @test Catalyst.issparse(A_k)
 
-let
-    rn = @reaction_network begin
-        (k1, k2), 2A <--> D
-        (k3, k4), D <--> A + B
-        (k5, k6), A + B <--> C
-        (k7, k8), C <--> 2B
-        (k9, k10), C + D <--> E + F
-        (k11, k12), E + F <--> H
-        (k13, k14), H <--> C + E
-        (k15, k16), C + E <--> D + F
-        (k17, k18), A + D <--> G
-        (k19, k20), G <--> B + H
+    numeqs = similar(eqs)
+    for i in 1:length(eqs)
+        numeqs[i] = substitute(eqs[i], ratemap)
     end
-
-    @test Catalyst.satisfiesdeficiencyone(rn) == true
+    @test all(iszero, simplify(numeqs - S*K*Φ))
+    @test all(iszero, simplify(numeqs - Y*A_k*Φ))
 end
 
-### Some tests for deficiency zero networks. 
-
+# Test handling for weird complexes. 
 let
     rn = @reaction_network begin
-        (k1, k2), A <--> 2B
-        (k3, k4), A + C <--> D
-        k5, D --> B + E
-        k6, B + E --> A + C
-    end
-
-    # No longer weakly reversible
-    rn2 = @reaction_network begin
-        (k1, k2), A <--> 2B
-        (k3, k4), A + C <--> D
-        k5, B + E --> D
-        k6, B + E --> A + C
-    end
-
-    # No longer weakly reversible
-    rn3 = @reaction_network begin
-        k1, A --> 2B
-        (k3, k4), A + C <--> D
-        k5, D --> B + E 
-        k6, B + E --> A + C
+        k1, 2X + Y + 3Z --> ∅
```

Comment:
Add tests that the ODEs from the new decomposition are the same as what Catalyst generates in each case of `combinatoric_ratelaws`. I suspect right now they aren't the same.